src/stability_sdk/animation.py

```diff
@@ -173,27 +175,82 @@ class AnimationArgs(
     VideoInputSettings,
     VideoOutputSettings
 ):
-    pass
-
+    """
+    Aggregates parameters from the multiple settings classes.
+    """
 
-def args2dict(args):
+@dataclass
+class FrameArgs:
+    """Expansion of key framed Args to per-frame values"""
+    angle: List[float]
+    zoom: List[float]
+    translation_x: List[float]
+    translation_y: List[float]
+    translation_z: List[float]
+    rotation_x: List[float]
+    rotation_y: List[float]
+    rotation_z: List[float]
+    brightness_curve: List[float]
+    contrast_curve: List[float]
+    hue_curve: List[float]
+    saturation_curve: List[float]
+    lightness_curve: List[float]
+    noise_add_curve: List[float]
+    noise_scale_curve: List[float]
+    steps_curve: List[float]
```

Comment:
Do we want this one to be `float`?

Reply:
Good question. For uniformity and simplicity I favor it. 

Where it gets used: `steps = int(self.frame_args.steps_curve[frame_idx])` (same as before)

And where it gets expanded needs no special case when all animation curves considered float
```
        # expand key frame strings to per frame series
        frame_args_dict = {f.name: curve_to_series(getattr(args, f.name)) for f in fields(FrameArgs)}
        self.frame_args = FrameArgs(**frame_args_dict) 
```

The `diffusion_cadence_curve` is the only other animated value that might be considered for int type.